subcommand/health-sync/checks.go

```diff
@@ -161,8 +161,37 @@ func (c *Command) syncChecks(consulClient *api.Client,
 				currentStatuses[container.Name] = container.Health.Status
 			}
 		}
+		if container.Name == config.ConsulDataplaneContainerName {
+			if container.Health.Status == ecs.HealthStatusHealthy {
+				overallDataplaneHealth = 1
+			}
+		}
```

Comment:
If I understand right, what we are trying to achieve here is:
- if any container health is unhealthy (including dataplane) we would like to set the dataplane health to unhealthy
- otherwise dataplane health is healthy
is that right?

If so the special case in here for the specific dataplane container is not needed. Also see my suggestion below